File: algorithms/1742.maximum-number-of-balls-in-a-box/solution.go
func countBalls(lowLimit int, highLimit int) int {
	m := map[int]int{}

	for i := lowLimit; i < highLimit+1; i++ {
		sum := digitsSum(i)
		if _, ok := m[sum]; ok {
			m[sum] += 1
		} else {
			m[sum] = 1
		}
	}

	max := 0
	for _, v := range m {
		if v > max {
			max = v
		}
	}

	return max
}

func digitsSum(num int) int {
	sum := 0

	for _, s := range strings.Split(strconv.Itoa(num), "") {
		i, _ := strconv.Atoi(s)
		sum += i
	}

	return sum
}

File: algorithms/1742.maximum-number-of-balls-in-a-box/solution.rs
impl Solution {
    pub fn count_balls(low_limit: i32, high_limit: i32) -> i32 {
        use std::cmp;
        use std::collections::HashMap;

        let mut map: HashMap<i32, i32> = HashMap::new();

        for num in low_limit..=high_limit {
            let sum = Self::digits_sum(num);
            let count = map.entry(sum).or_insert(0);
            *count += 1
        }

        let mut max = 0;
        for (_, v) in map {
            max = cmp::max(v, max)
        }
        max
    }

    fn digits_sum(num: i32) -> i32 {
        num.to_string().chars().map(|c| c as i32 - 48).sum()
    }
}
